### ISSUE TITLE
have server log client requests

### ISSUE THREAD

**Issue by taybin** [rethinkdb/rethinkdb#544]
It would be good if there was a more verbose logging option. Right now, I haven't seen rethinkdb print out anything after starting up.

**Comment by flockonus** [comment]
+1 fww - my motivation to use this feature is to debug development code. The JS library doesn't seem to have a way to log queries.

**Comment by Tryneus** [comment]
Now that we have the `rethinkdb.jobs` table, we can add that information to it with #3636.  Then, if we make changefeeds on the `jobs` table not poll (and potentially miss short-running queries), it would be trivial to implement a query log in an arbitrary client using:

```py
r.db('rethinkdb').table('jobs').filter(r.row['id'][0].eq('query')).changes(squash=False)
```

We could even integrate it into the Web UI or data explorer.  Of course, we might want to filter out the web UI's requests to avoid spam.

**Comment by danielmewes** [comment]
Yeah that would be pretty cool. We'll also need a query pretty printer, probably based on https://github.com/rethinkdb/rethinkdb/issues/3636

**Comment by danielmewes** [comment]
I'd like to consider this for 2.2, since it would be very useful in combination with access permissions (https://github.com/rethinkdb/rethinkdb/issues/4519 ) for monitoring / auditing purposes.

**Comment by coffeemug** [comment]
FYI, talked to a user today in a regulated industry; audit trails require logging actual data changes, along with the user information.

**Comment by marshall007** [comment]
Is it currently possible to filter the output of `r.db('rethinkdb').table('jobs').changes()` to only include jobs from the current client or do we have to wait on something like #4519?

The `info.client_port` property appears to be unique across clients, but is it possible to obtain that information from the drivers?

**Comment by coffeemug** [comment]
Unfortunately I can't think of a way of doing that right now (AFAIK the server doesn't send the `client_port` back to the drivers). We should definitely make this information available in the drivers, it would make the property in the `jobs` table dramatically more useful.

**Comment by danielmewes** [comment]
I opened https://github.com/rethinkdb/rethinkdb/issues/4620 for the `client_port` suggestion.

**Comment by Omeryl** [comment]
This would be really, really helpful :).

**Comment by shinzui** [comment]
I am curious how to debug bad clients without any logging. I tried using a changefeed filtered to my database but that was not very helpful. This issues is three years old so there must be a workaround that I am missing.

**Comment by mxstbr** [comment]
Has anybody found a workaround for this yet? It's been a couple of years now... :wink:

**Comment by edoardodelbino** [comment]
Any workaround to log client requests? If this thread has been dropped is it because you have all found alternative solutions?

**Comment by mishawakerman** [comment]
I'd love to re-open this discussion.

We ❤️ and use Rethink to deliver an enterprise SaaS application and an increasing number of our customers are unhappy that we cannot provide audit-able logs for all data accesses and modifications **including/especially those done via the Data Explorer.** 

As discussed above, a query logger that includes clients and their connections/queries would be great. If the log files can be automatically rotated by date that would be a bonus.

I understand the "active" status of development here is still pending developers/funding but we're open to funding this development or providing some engineering time if some guidance can be offered by someone with experience here.

**Comment by srh** [comment]
Hey @mishawakerman .  I believe RethinkDB Enterprise has audit logging implemented.  There's some chance it's just in-progress, I don't know.  The repository is at https://github.com/rethinkdb/rethinkdb-enterprise/ .

I'm unwilling to look at it unless the repo explicitly gets open sourced.  As far as I'm concerned, anything in there is proprietary code owned by the Linux Foundation.

So we might want to chase down whoever's in charge of that sort of thing (@gabor-boros ?) and get it open sourced.  Then, integrating audit logging on top of 2.4.x (or you just using the repo as-is) is a possibility.

**Comment by gabor-boros** [comment]
@srh My first thing tomorrow morning will be to contact LF

**Comment by mishawakerman** [comment]
Thanks for the prompt replies @srh and @gabor-boros. Interested in knowing the next steps.

**Comment by gabor-boros** [comment]
@mishawakerman, @srh I just contacted LF and waiting for their reply. I would say that rethinkdb-enterprise should be Apache 2.0 licensed as well and we forgot to replace the license info, but I'm not sure. Let's wait for them to get back to me.

**Comment by gabor-boros** [comment]
I got the reply from LF. The rethinkdb-enterprise has been relicensed under Apache 2.0; dependencies, of course, remain under their own license.

@srh You can look at it, it is explicitly open sourced. Also, there is an action item for me: Replace the weird license within the repo to Apache 2.0.

**Comment by srh** [comment]
Great!  Thanks @gabor-boros .

**Comment by gabor-boros** [comment]
@srh do you know about any great features implemented there what we could move as well? In my opinion the enterprise version doesn’t make sense now and every cool stuff did there could be moved. What do you think?

**Comment by srh** [comment]
I'm not familiar with its feature set.  I think there is one other big feature, something to do with outdated reads.
